src/externalized/mercs/MercProfile.cc

```diff
@@ -90,4 +91,409 @@ MERCPROFILESTRUCT &MercProfile::getStruct() const
+		prof->sSector = SGPSector::FromShortString(jShort);
+		if (jLong.size() == 2) {
+			prof->sSector.z = jLong[1].to_int();
+		}
+	}
+
+	prof->bTown = Internals::getTownEnumFromString(json.getOptionalString("town"));
+	prof->bTownAttachment = json.getOptionalUInt("townAttachment");
+	if (json.getOptionalBool("isTownIndifferentIfDead")) {
+		prof->ubMiscFlags3 |= PROFILE_MISC_FLAG3_TOWN_DOESNT_CARE_ABOUT_DEATH;
+	}
+	if (json.has("ownedRooms")) {
+		JsonObject rooms = json["ownedRooms"].toObject();
+		if (rooms.has("range1")) {
+			std::vector<JsonValue> range1 = rooms["range1"].toVec();
+			prof->ubRoomRangeStart[0] = range1[0].toInt();
+			prof->ubRoomRangeEnd[0] = range1[1].toInt();
+		}
+		if (rooms.has("range2")) {
+			std::vector<JsonValue> range2 = rooms["range2"].toVec();
+			prof->ubRoomRangeStart[1] = range2[0].toInt();
+			prof->ubRoomRangeEnd[1] = range2[1].toInt();
+		}
+	}
+
+	prof->bReputationTolerance = json.getOptionalUInt("toleranceForPlayersReputation", 101);
+	prof->bDeathRate = json.getOptionalUInt("toleranceForPlayersDeathRate", 101);
+	if (json.has("contract")) {
+		JsonObject c = json["contract"].toObject();
+		prof->sSalary = c.getOptionalUInt("dailySalary");
+		prof->uiWeeklySalary = c.getOptionalUInt("weeklySalary");
+		prof->uiBiWeeklySalary = c.getOptionalUInt("biWeeklySalary");
+		prof->bMedicalDeposit = c.getOptionalBool("isMedicalDepositRequired");
+	}
+
+	if (json.has("inventory")) {
+		for (auto& element : json.GetValue("inventory").toVec()) {
+			JsonObject s = element.toObject();
+			InvSlotPos slotIdx = Internals::getInventorySlotEnumFromString(s.GetString("slot"));
+			prof->inv[slotIdx] = contentManager->getItemByName(s.GetString("item"))->getItemIndex();
+			prof->bInvNumber[slotIdx] = s.getOptionalUInt("quantity", 1);
+			prof->bInvStatus[slotIdx] = s.getOptionalUInt("status", 100);
+			if (s.getOptionalBool("isUndroppable")) {
+				prof->ubInvUndroppable |= gubItemDroppableFlag[slotIdx];
+			}
+		}
+	}
+	prof->uiMoney = json.getOptionalUInt("money");
+
+	if (json.has("dialogue")) {
+		for (auto& element : json.GetValue("dialogue").toVec()) {
+			JsonObject a = element.toObject();
+			Approach jApproach = Internals::getApproachEnumFromString(a.GetString("approach"));
+			if (jApproach > 0 && jApproach < 5) {
+				prof->usApproachFactor[jApproach - 1] = a.getOptionalUInt("effectiveness");
+				prof->ubApproachVal[jApproach - 1] = a.getOptionalUInt("desireToTalk");
+				prof->ubApproachMod[APPROACH_FRIENDLY - 1][jApproach - 1] = a.getOptionalUInt("friendlyMod");
+				prof->ubApproachMod[APPROACH_DIRECT - 1][jApproach - 1] = a.getOptionalUInt("directMod");
+				prof->ubApproachMod[APPROACH_THREATEN - 1][jApproach - 1] = a.getOptionalUInt("threatenMod");
+			}
+		}
+	}
+
+	return prof;
+}
+
+JsonValue MercProfile::serializeStruct(const ContentManager* contentManager) const
+{
+	MERCPROFILESTRUCT* prof = this->m_profile;
+	JsonObject obj;
+	obj.set("000profile", this->getInfo().internalName);
```

Comment:
Why "000profile", etc. instead of just "profile"?

Reply:
This is a workaround to place fields logically grouped instead of alphabetically. I forgot to add a comment explaining this.